godot-core/src/engine.rs

```diff
@@ -186,21 +186,24 @@ where
 // Utilities for crate
 
 pub(crate) fn debug_string<T: GodotClass>(
-    ptr: &Gd<T>,
+    ptr: &RawGd<T>,
     f: &mut std::fmt::Formatter<'_>,
     ty: &str,
 ) -> std::fmt::Result {
-    if let Some(id) = ptr.instance_id_or_none() {
+    if ptr.is_instance_valid() {
+        let id = ptr.instance_id().unwrap();
```

Comment:
Maybe `instance_id_or_none` should be re-introduced for better encapsulation. Would also avoid the double `if` (in `is_instance_valid` and in `unwrap`).